sympy/matrices/expressions/matexpr.py

```diff
@@ -436,7 +442,7 @@ def from_index_summation(expr, first_index=None, last_index=None, dimensions=Non
         >>> MatrixExpr.from_index_summation(expr)
         A*B.T*A.T
         """
-        from sympy import Sum, Mul, Add, MatMul, transpose, trace
+        from sympy import Sum, transpose, trace
```

Comment:
I would expect things like this to be picked up by flake8. Is there maybe a setting that should be enabled?

Reply:
I use Spyder's built-in which relies on PyLint. It seems like flake8 doesn't support this type of violations: https://flake8.pycqa.org/en/latest/user/error-codes.html

Worth noting though that PyLint often reports errors in the SymPy code base, even though the code is working. But I guess it is a matter of filtering. This particular one corresponds to W0404 (reimported) and W0621 (redefined-outer-name). MatMul only gets this, but that is because it is imported at the end of the file.

Reply:
There are a few of these... (Partial results)
```
************* Module sympy.assumptions.refine
sympy\sympy\assumptions\refine.py:235:4: W0404: Reimport 'S' (imported line 3) (reimported)
************* Module sympy.assumptions.tests.test_query
sympy\sympy\assumptions\tests\test_query.py:2325:12: W0404: Reimport 'log' (imported line 18) (reimported)
sympy\sympy\assumptions\tests\test_query.py:2350:12: W0404: Reimport 'log' (imported line 18) (reimported)
************* Module sympy.assumptions.tests.test_refine
sympy\sympy\assumptions\tests\test_refine.py:189:4: W0404: Reimport 'Expr' (imported line 1) (reimported)
************* Module sympy.calculus.util
sympy\sympy\calculus\util.py:66:4: W0404: Reimport 'solve_univariate_inequality' (imported line 15) (reimported)
************* Module sympy.calculus.tests.test_util
sympy\sympy\calculus\tests\test_util.py:194:4: W0404: Reimport 'pi' (imported line 1) (reimported)
************* Module sympy.combinatorics.fp_groups
sympy\sympy\combinatorics\fp_groups.py:233:8: W0404: Reimport 'S' (imported line 3) (reimported)
sympy\sympy\combinatorics\fp_groups.py:382:8: W0404: Reimport 'PermutationGroup' (imported line 10) (reimported)
sympy\sympy\combinatorics\fp_groups.py:689:8: W0404: Reimport 'S' (imported line 3) (reimported)
************* Module sympy.combinatorics.homomorphisms
sympy\sympy\combinatorics\homomorphisms.py:119:8: W0404: Reimport 'S' (imported line 7) (reimported)
sympy\sympy\combinatorics\homomorphisms.py:202:8: W0404: Reimport 'S' (imported line 7) (reimported)
************* Module sympy.combinatorics.testutil
sympy\sympy\combinatorics\testutil.py:243:4: W0404: Reimport 'Permutation' (imported line 1) (reimported)
************* Module sympy.combinatorics.tests.test_fp_groups
sympy\sympy\combinatorics\tests\test_fp_groups.py:199:4: W0404: Reimport 'FpSubgroup' (imported line 2) (reimported)
************* Module sympy.combinatorics.tests.test_permutations
sympy\sympy\combinatorics\tests\test_permutations.py:476:4: W0404: Reimport 'sstr' (imported line 10) (reimported)
sympy\sympy\combinatorics\tests\test_permutations.py:476:4: W0404: Reimport 'srepr' (imported line 10) (reimported)
************* Module sympy.combinatorics.tests.test_polyhedron
sympy\sympy\combinatorics\tests\test_polyhedron.py:88:4: W0404: Reimport 'tetrahedron' (imported line 2) (reimported)
sympy\sympy\combinatorics\tests\test_polyhedron.py:88:4: W0404: Reimport 'cube' (imported line 2) (reimported)
sympy\sympy\combinatorics\tests\test_polyhedron.py:88:4: W0404: Reimport 'octahedron' (imported line 2) (reimported)
sympy\sympy\combinatorics\tests\test_polyhedron.py:88:4: W0404: Reimport 'dodecahedron' (imported line 2) (reimported)
sympy\sympy\combinatorics\tests\test_polyhedron.py:88:4: W0404: Reimport 'icosahedron' (imported line 2) (reimported)
sympy\sympy\combinatorics\tests\test_polyhedron.py:88:4: W0404: Reimport 'cube_faces' (imported line 2) (reimported)
************* Module sympy.concrete.summations
sympy\sympy\concrete\summations.py:325:8: W0404: Reimport 'Mul' (imported line 8) (reimported)
sympy\sympy\concrete\summations.py:1039:4: W0404: Reimport 'Add' (imported line 6) (reimported)
sympy\sympy\concrete\summations.py:1290:4: W0404: Reimport 'And' (imported line 15) (reimported)
************* Module sympy.concrete.tests.test_sums_products
sympy\sympy\concrete\tests\test_sums_products.py:642:4: W0404: Reimport 'sin' (imported line 1) (reimported)
sympy\sympy\concrete\tests\test_sums_products.py:1433:4: W0404: Reimport 'floor' (imported line 1) (reimported)
sympy\sympy\concrete\tests\test_sums_products.py:1433:4: W0404: Reimport 'Sum' (imported line 1) (reimported)
sympy\sympy\concrete\tests\test_sums_products.py:1433:4: W0404: Reimport 'Symbol' (imported line 1) (reimported)
************* Module sympy.core.exprtools
sympy\sympy\core\exprtools.py:1357:8: W0404: Reimport 'Dummy' (imported line 12) (reimported)
************* Module sympy.core.numbers
sympy\sympy\core\numbers.py:176:4: W0404: Reimport 'MPZ' (imported line 25) (reimported)
************* Module sympy.core.tests.test_args
sympy\sympy\core\tests\test_args.py:586:4: W0404: Reimport 'Basic' (imported line 10) (reimported)
sympy\sympy\core\tests\test_args.py:756:4: W0404: Reimport 'Rational' (imported line 10) (reimported)
sympy\sympy\core\tests\test_args.py:859:4: W0404: Reimport 'S' (imported line 10) (reimported)
sympy\sympy\core\tests\test_args.py:879:4: W0404: Reimport 'Interval' (imported line 10) (reimported)
sympy\sympy\core\tests\test_args.py:884:4: W0404: Reimport 'Interval' (imported line 10) (reimported)
sympy\sympy\core\tests\test_args.py:895:4: W0404: Reimport 'Interval' (imported line 10) (reimported)
sympy\sympy\core\tests\test_args.py:905:4: W0404: Reimport 'Interval' (imported line 10) (reimported)
sympy\sympy\core\tests\test_args.py:967:4: W0404: Reimport 'S' (imported line 10) (reimported)
sympy\sympy\core\tests\test_args.py:968:4: W0404: Reimport 'Interval' (imported line 10) (reimported)
sympy\sympy\core\tests\test_args.py:978:4: W0404: Reimport 'Interval' (imported line 10) (reimported)
sympy\sympy\core\tests\test_args.py:986:4: W0404: Reimport 'S' (imported line 10) (reimported)
sympy\sympy\core\tests\test_args.py:987:4: W0404: Reimport 'Interval' (imported line 10) (reimported)
sympy\sympy\core\tests\test_args.py:995:4: W0404: Reimport 'S' (imported line 10) (reimported)
sympy\sympy\core\tests\test_args.py:1007:4: W0404: Reimport 'S' (imported line 10) (reimported)
sympy\sympy\core\tests\test_args.py:1169:4: W0404: Reimport 'S' (imported line 10) (reimported)
sympy\sympy\core\tests\test_args.py:1358:4: W0404: Reimport 'Interval' (imported line 10) (reimported)
sympy\sympy\core\tests\test_args.py:1358:4: W0404: Reimport 'Lambda' (imported line 10) (reimported)
sympy\sympy\core\tests\test_args.py:1366:4: W0404: Reimport 'Lambda' (imported line 10) (reimported)
sympy\sympy\core\tests\test_args.py:1527:4: W0404: Reimport 'NormalDistribution' (imported line 1021) (reimported)
sympy\sympy\core\tests\test_args.py:1723:4: W0404: Reimport 'MatrixSymbol' (imported line 3242) (reimported)
sympy\sympy\core\tests\test_args.py:1729:4: W0404: Reimport 'MatrixSymbol' (imported line 3242) (reimported)
sympy\sympy\core\tests\test_args.py:1740:4: W0404: Reimport 'MatrixSymbol' (imported line 3242) (reimported)
sympy\sympy\core\tests\test_args.py:1745:4: W0404: Reimport 'MatrixSymbol' (imported line 3242) (reimported)
sympy\sympy\core\tests\test_args.py:1849:4: W0404: Reimport 'MatrixSymbol' (imported line 3242) (reimported)
sympy\sympy\core\tests\test_args.py:1857:4: W0404: Reimport 'MatrixSymbol' (imported line 3242) (reimported)
sympy\sympy\core\tests\test_args.py:2065:4: W0404: Reimport 'exp' (imported line 10) (reimported)
sympy\sympy\core\tests\test_args.py:2075:4: W0404: Reimport 'log' (imported line 10) (reimported)
sympy\sympy\core\tests\test_args.py:3047:4: W0404: Reimport 'MatrixSymbol' (imported line 3242) (reimported)
sympy\sympy\core\tests\test_args.py:3054:4: W0404: Reimport 'MatrixSymbol' (imported line 3242) (reimported)
sympy\sympy\core\tests\test_args.py:3062:4: W0404: Reimport 'MatrixSymbol' (imported line 3242) (reimported)
sympy\sympy\core\tests\test_args.py:3070:4: W0404: Reimport 'MatrixSymbol' (imported line 3242) (reimported)
sympy\sympy\core\tests\test_args.py:3080:4: W0404: Reimport 'MatrixSymbol' (imported line 3242) (reimported)
sympy\sympy\core\tests\test_args.py:3086:4: W0404: Reimport 'MatrixSymbol' (imported line 3242) (reimported)
sympy\sympy\core\tests\test_args.py:3097:4: W0404: Reimport 'MatrixSymbol' (imported line 3242) (reimported)
sympy\sympy\core\tests\test_args.py:3098:4: W0404: Reimport 'S' (imported line 10) (reimported)
sympy\sympy\core\tests\test_args.py:3102:4: W0404: Reimport 'MatrixSymbol' (imported line 3242) (reimported)
sympy\sympy\core\tests\test_args.py:3133:4: W0404: Reimport 'S' (imported line 10) (reimported)
sympy\sympy\core\tests\test_args.py:3138:4: W0404: Reimport 'MatrixSymbol' (imported line 3242) (reimported)
sympy\sympy\core\tests\test_args.py:3146:4: W0404: Reimport 'MatrixSymbol' (imported line 3242) (reimported)
sympy\sympy\core\tests\test_args.py:3153:4: W0404: Reimport 'MatrixSymbol' (imported line 3242) (reimported)
sympy\sympy\core\tests\test_args.py:3159:4: W0404: Reimport 'MatrixSymbol' (imported line 3242) (reimported)
sympy\sympy\core\tests\test_args.py:3165:4: W0404: Reimport 'MatrixSymbol' (imported line 3242) (reimported)
sympy\sympy\core\tests\test_args.py:3171:4: W0404: Reimport 'MatrixSymbol' (imported line 3242) (reimported)
sympy\sympy\core\tests\test_args.py:3178:4: W0404: Reimport 'MatrixSymbol' (imported line 3242) (reimported)
sympy\sympy\core\tests\test_args.py:3186:4: W0404: Reimport 'MatrixSymbol' (imported line 3242) (reimported)
sympy\sympy\core\tests\test_args.py:3194:4: W0404: Reimport 'MatrixSymbol' (imported line 3242) (reimported)
sympy\sympy\core\tests\test_args.py:3201:4: W0404: Reimport 'MatrixSymbol' (imported line 3242) (reimported)
sympy\sympy\core\tests\test_args.py:3207:4: W0404: Reimport 'MatrixSymbol' (imported line 3242) (reimported)
sympy\sympy\core\tests\test_args.py:3213:4: W0404: Reimport 'MatrixSymbol' (imported line 3242) (reimported)
sympy\sympy\core\tests\test_args.py:3218:4: W0404: Reimport 'MatrixSymbol' (imported line 3242) (reimported)
sympy\sympy\core\tests\test_args.py:3223:4: W0404: Reimport 'MatrixSymbol' (imported line 3242) (reimported)
sympy\sympy\core\tests\test_args.py:3228:4: W0404: Reimport 'symbols' (imported line 10) (reimported)
sympy\sympy\core\tests\test_args.py:3234:4: W0404: Reimport 'S' (imported line 10) (reimported)
sympy\sympy\core\tests\test_args.py:3239:4: W0404: Reimport 'S' (imported line 10) (reimported)
sympy\sympy\core\tests\test_args.py:3300:4: W0404: Reimport 'MatrixSymbol' (imported line 3242) (reimported)
sympy\sympy\core\tests\test_args.py:3387:4: W0404: Reimport 'S' (imported line 10) (reimported)
sympy\sympy\core\tests\test_args.py:3543:4: W0404: Reimport 'oo' (imported line 10) (reimported)
sympy\sympy\core\tests\test_args.py:3543:4: W0404: Reimport 'Interval' (imported line 10) (reimported)
sympy\sympy\core\tests\test_args.py:3569:4: W0404: Reimport 'Function' (imported line 10) (reimported)
sympy\sympy\core\tests\test_args.py:3946:4: W0404: Reimport 'sin' (imported line 10) (reimported)
sympy\sympy\core\tests\test_args.py:5048:4: W0404: Reimport 'MatrixSymbol' (imported line 3242) (reimported)
************* Module sympy.core.tests.test_basic
sympy\sympy\core\tests\test_basic.py:240:4: W0404: Reimport 'Integral' (imported line 14) (reimported)
sympy\sympy\core\tests\test_basic.py:241:4: W0404: Reimport 'symbols' (imported line 9) (reimported)
sympy\sympy\core\tests\test_basic.py:242:4: W0404: Reimport 'Function' (imported line 11) (reimported)
sympy\sympy\core\tests\test_basic.py:243:4: W0404: Reimport 'cos' (imported line 14) (reimported)
sympy\sympy\core\tests\test_basic.py:243:4: W0404: Reimport 'sin' (imported line 14) (reimported)
************* Module sympy.core.tests.test_evalf
sympy\sympy\core\tests\test_evalf.py:379:4: W0404: Reimport 'I' (imported line 1) (reimported)
************* Module sympy.core.tests.test_expr
sympy\sympy\core\tests\test_expr.py:352:4: W0404: Reimport 'Gt' (imported line 1) (reimported)
sympy\sympy\core\tests\test_expr.py:1866:4: W0404: Reimport 'x' (imported line 16) (reimported)
************* Module sympy.core.tests.test_exprtools
sympy\sympy\core\tests\test_exprtools.py:452:4: W0404: Reimport 'Symbol' (imported line 3) (reimported)
sympy\sympy\core\tests\test_exprtools.py:452:4: W0404: Reimport 'Sum' (imported line 3) (reimported)
sympy\sympy\core\tests\test_exprtools.py:469:4: W0404: Reimport 'gcd_terms' (imported line 6) (reimported)
************* Module sympy.core.tests.test_function
sympy\sympy\core\tests\test_function.py:575:4: W0404: Reimport 'polygamma' (imported line 1) (reimported)
sympy\sympy\core\tests\test_function.py:894:4: W0404: Reimport '_aresame' (imported line 6) (reimported)
sympy\sympy\core\tests\test_function.py:990:4: W0404: Reimport 'Function' (imported line 1) (reimported)
sympy\sympy\core\tests\test_function.py:991:4: W0404: Reimport 'x' (imported line 20) (reimported)
sympy\sympy\core\tests\test_function.py:1003:4: W0404: Reimport 'Function' (imported line 1) (reimported)
sympy\sympy\core\tests\test_function.py:1014:4: W0404: Reimport 'x' (imported line 20) (reimported)
************* Module sympy.core.tests.test_match
sympy\sympy\core\tests\test_match.py:696:4: W0404: Reimport 'symbols' (imported line 1) (reimported)
sympy\sympy\core\tests\test_match.py:696:4: W0404: Reimport 'Wild' (imported line 1) (reimported)
************* Module sympy.core.tests.test_numbers
sympy\sympy\core\tests\test_numbers.py:601:4: W0404: Reimport 'mpmath' (imported line 20) (reimported)
sympy\sympy\core\tests\test_numbers.py:1409:4: W0404: Reimport 'pi' (imported line 3) (reimported)
************* Module sympy.core.tests.test_relational
sympy\sympy\core\tests\test_relational.py:21:4: W0404: Reimport 'raises' (imported line 4) (reimported)
sympy\sympy\core\tests\test_relational.py:266:4: W0404: Reimport 'Symbol' (imported line 5) (reimported)
sympy\sympy\core\tests\test_relational.py:1149:4: W0404: Reimport 'Expr' (imported line 5) (reimported)
************* Module sympy.core.tests.test_symbol
sympy\sympy\core\tests\test_symbol.py:56:4: W0404: Reimport 'sympify' (imported line 1) (reimported)
************* Module sympy.functions.combinatorial.numbers
sympy\sympy\functions\combinatorial\numbers.py:1164:8: W0404: Reimport 'log' (imported line 20) (reimported)
sympy\sympy\functions\combinatorial\numbers.py:1516:4: W0404: Reimport 'factorial' (imported line 19) (reimported)
sympy\sympy\functions\combinatorial\numbers.py:1687:4: W0404: Reimport 'binomial' (imported line 19) (reimported)
************* Module sympy.functions.combinatorial.tests.test_comb_factorials
sympy\sympy\functions\combinatorial\tests\test_comb_factorials.py:497:4: W0404: Reimport 'I' (imported line 1) (reimported)
************* Module sympy.functions.combinatorial.tests.test_comb_numbers
sympy\sympy\functions\combinatorial\tests\test_comb_numbers.py:482:4: W0404: Reimport 'oo' (imported line 3) (reimported)
************* Module sympy.functions.elementary.complexes
sympy\sympy\functions\elementary\complexes.py:1220:8: W0404: Reimport 'oo' (imported line 7) (reimported)
sympy\sympy\functions\elementary\complexes.py:1220:8: W0404: Reimport 'exp_polar' (imported line 9) (reimported)
sympy\sympy\functions\elementary\complexes.py:1220:8: W0404: Reimport 'I' (imported line 7) (reimported)
sympy\sympy\functions\elementary\complexes.py:1220:8: W0404: Reimport 'Mul' (imported line 1) (reimported)
sympy\sympy\functions\elementary\complexes.py:1220:8: W0404: Reimport 'Symbol' (imported line 1) (reimported)
sympy\sympy\functions\elementary\complexes.py:1271:8: W0404: Reimport 'exp' (imported line 9) (reimported)
sympy\sympy\functions\elementary\complexes.py:1271:8: W0404: Reimport 'pi' (imported line 7) (reimported)
sympy\sympy\functions\elementary\complexes.py:1271:8: W0404: Reimport 'I' (imported line 7) (reimported)
************* Module sympy.functions.elementary.exponential
sympy\sympy\functions\elementary\exponential.py:480:8: W0404: Reimport 'Wild' (imported line 13) (reimported)
************* Module sympy.functions.elementary.miscellaneous
sympy\sympy\functions\elementary\miscellaneous.py:437:8: W0404: Reimport 'ordered' (imported line 5) (reimported)
************* Module sympy.functions.elementary.piecewise
sympy\sympy\functions\elementary\piecewise.py:537:8: W0404: Reimport 'Dummy' (imported line 1) (reimported)
sympy\sympy\functions\elementary\piecewise.py:1090:8: W0404: Reimport 'sift' (imported line 9) (reimported)
************* Module sympy.functions.elementary.trigonometric
sympy\sympy\functions\elementary\trigonometric.py:451:8: W0404: Reimport 'expand_mul' (imported line 6) (reimported)
sympy\sympy\functions\elementary\trigonometric.py:2234:8: W0404: Reimport 'log' (imported line 13) (reimported)
sympy\sympy\functions\elementary\trigonometric.py:2437:8: W0404: Reimport 'log' (imported line 13) (reimported)
sympy\sympy\functions\elementary\trigonometric.py:3034:8: W0404: Reimport 'log' (imported line 13) (reimported)
sympy\sympy\functions\elementary\trigonometric.py:3206:8: W0404: Reimport 'log' (imported line 13) (reimported)
************* Module sympy.functions.elementary.tests.test_complexes
sympy\sympy\functions\elementary\tests\test_complexes.py:777:4: W0404: Reimport 'exp_polar' (imported line 1) (reimported)
sympy\sympy\functions\elementary\tests\test_complexes.py:777:4: W0404: Reimport 'exp' (imported line 1) (reimported)
sympy\sympy\functions\elementary\tests\test_complexes.py:779:4: W0404: Reimport 'sin' (imported line 1) (reimported)
sympy\sympy\functions\elementary\tests\test_complexes.py:779:4: W0404: Reimport 'Eq' (imported line 1) (reimported)
sympy\sympy\functions\elementary\tests\test_complexes.py:856:4: W0404: Reimport 'Matrix' (imported line 1) (reimported)
sympy\sympy\functions\elementary\tests\test_complexes.py:870:4: W0404: Reimport 'oo' (imported line 1) (reimported)
sympy\sympy\functions\elementary\tests\test_complexes.py:870:4: W0404: Reimport 'pi' (imported line 1) (reimported)
sympy\sympy\functions\elementary\tests\test_complexes.py:908:4: W0404: Reimport 'exp_polar' (imported line 1) (reimported)
************* Module sympy.functions.elementary.tests.test_exponential
sympy\sympy\functions\elementary\tests\test_exponential.py:166:4: W0404: Reimport 'Sum' (imported line 1) (reimported)
sympy\sympy\functions\elementary\tests\test_exponential.py:649:4: W0404: Reimport 'Product' (imported line 1) (reimported)
sympy\sympy\functions\elementary\tests\test_exponential.py:684:4: W0404: Reimport 'Product' (imported line 1) (reimported)
sympy\sympy\functions\elementary\tests\test_exponential.py:684:4: W0404: Reimport 'Sum' (imported line 1) (reimported)
************* Module sympy.functions.elementary.tests.test_piecewise
sympy\sympy\functions\elementary\tests\test_piecewise.py:686:4: W0404: Reimport 'sin' (imported line 1) (reimported)
sympy\sympy\functions\elementary\tests\test_piecewise.py:686:4: W0404: Reimport 'cos' (imported line 1) (reimported)
************* Module sympy.functions.elementary.tests.test_trigonometric
sympy\sympy\functions\elementary\tests\test_trigonometric.py:200:4: W0404: Reimport 'unchanged' (imported line 7) (reimported)
************* Module sympy.functions.special.bessel
sympy\sympy\functions\special\bessel.py:1080:4: W0404: Reimport 'pi' (imported line 3) (reimported)
sympy\sympy\functions\special\bessel.py:1872:8: W0404: Reimport 'exp' (imported line 11) (reimported)
sympy\sympy\functions\special\bessel.py:1896:8: W0404: Reimport 'exp' (imported line 11) (reimported)
sympy\sympy\functions\special\bessel.py:1906:8: W0404: Reimport 'exp' (imported line 11) (reimported)
sympy\sympy\functions\special\bessel.py:1912:8: W0404: Reimport 'exp' (imported line 11) (reimported)
sympy\sympy\functions\special\bessel.py:1918:12: W0404: Reimport 'exp' (imported line 11) (reimported)
************* Module sympy.functions.special.error_functions
sympy\sympy\functions\special\error_functions.py:1345:8: W0404: Reimport 'exp' (imported line 11) (reimported)
sympy\sympy\functions\special\error_functions.py:1345:8: W0404: Reimport 'factorial' (imported line 7) (reimported)
sympy\sympy\functions\special\error_functions.py:1367:8: W0404: Reimport 'meijerg' (imported line 15) (reimported)
sympy\sympy\functions\special\error_functions.py:1381:8: W0404: Reimport 'exp' (imported line 11) (reimported)
sympy\sympy\functions\special\error_functions.py:1381:8: W0404: Reimport 'factorial' (imported line 7) (reimported)
sympy\sympy\functions\special\error_functions.py:1779:8: W0404: Reimport 'log' (imported line 11) (reimported)
************* Module sympy.functions.special.hyper
sympy\sympy\functions\special\hyper.py:340:8: W0404: Reimport 're' (imported line 11) (reimported)
sympy\sympy\functions\special\hyper.py:340:8: W0404: Reimport 'oo' (imported line 4) (reimported)
************* Module sympy.functions.special.zeta_functions
sympy\sympy\functions\special\zeta_functions.py:123:8: W0404: Reimport 'I' (imported line 4) (reimported)
sympy\sympy\functions\special\zeta_functions.py:123:8: W0404: Reimport 'Add' (imported line 3) (reimported)
sympy\sympy\functions\special\zeta_functions.py:123:8: W0404: Reimport 'exp_polar' (imported line 8) (reimported)
sympy\sympy\functions\special\zeta_functions.py:340:8: W0404: Reimport 'log' (imported line 8) (reimported)
************* Module sympy.functions.special.tests.test_gamma_functions
sympy\sympy\functions\special\tests\test_gamma_functions.py:101:4: W0404: Reimport 'I' (imported line 1) (reimported)
sympy\sympy\functions\special\tests\test_gamma_functions.py:101:4: W0404: Reimport 'pi' (imported line 1) (reimported)
sympy\sympy\functions\special\tests\test_gamma_functions.py:101:4: W0404: Reimport 'exp_polar' (imported line 1) (reimported)
sympy\sympy\functions\special\tests\test_gamma_functions.py:111:4: W0404: Reimport 'exp_polar' (imported line 1) (reimported)
sympy\sympy\functions\special\tests\test_gamma_functions.py:111:4: W0404: Reimport 'I' (imported line 1) (reimported)
sympy\sympy\functions\special\tests\test_gamma_functions.py:180:4: W0404: Reimport 'exp_polar' (imported line 1) (reimported)
sympy\sympy\functions\special\tests\test_gamma_functions.py:180:4: W0404: Reimport 'I' (imported line 1) (reimported)
sympy\sympy\functions\special\tests\test_gamma_functions.py:228:4: W0404: Reimport 'I' (imported line 1) (reimported)
sympy\sympy\functions\special\tests\test_gamma_functions.py:301:4: W0404: Reimport 'exp_polar' (imported line 1) (reimported)
sympy\sympy\functions\special\tests\test_gamma_functions.py:397:4: W0404: Reimport 'I' (imported line 1) (reimported)
sympy\sympy\functions\special\tests\test_gamma_functions.py:525:4: W0404: Reimport 'I' (imported line 1) (reimported)
************* Module sympy.functions.special.tests.test_hyper
sympy\sympy\functions\special\tests\test_hyper.py:58:4: W0404: Reimport 'gamma' (imported line 1) (reimported)
************* Module sympy.functions.special.tests.test_zeta_functions
sympy\sympy\functions\special\tests\test_zeta_functions.py:144:4: W0404: Reimport 'log' (imported line 1) (reimported)
sympy\sympy\functions\special\tests\test_zeta_functions.py:175:4: W0404: Reimport 'verify_numerically' (imported line 7) (reimported)
************* Module sympy.geometry.ellipse
sympy\sympy\geometry\ellipse.py:1133:8: W0404: Reimport 'sin' (imported line 18) (reimported)
sympy\sympy\geometry\ellipse.py:1133:8: W0404: Reimport 'cos' (imported line 18) (reimported)
sympy\sympy\geometry\ellipse.py:1133:8: W0404: Reimport 'Rational' (imported line 13) (reimported)
************* Module sympy.geometry.line
sympy\sympy\geometry\line.py:2180:16: W0404: Reimport 'filldedent' (imported line 38) (reimported)
sympy\sympy\geometry\line.py:2622:8: W0404: Reimport 'filldedent' (imported line 38) (reimported)
************* Module sympy.geometry.plane
sympy\sympy\geometry\plane.py:864:8: W0404: Reimport 'solve' (imported line 16) (reimported)
************* Module sympy.geometry.util
sympy\sympy\geometry\util.py:288:4: W0404: Reimport 'sqrt' (imported line 14) (reimported)
************* Module sympy.holonomic.holonomic
sympy\sympy\holonomic\holonomic.py:2859:4: W0404: Reimport 'exp' (imported line 12) (reimported)
************* Module sympy.integrals.meijerint
sympy\sympy\integrals\meijerint.py:100:4: W0404: Reimport 'pi' (imported line 31) (reimported)
sympy\sympy\integrals\meijerint.py:100:4: W0404: Reimport 'cos' (imported line 47) (reimported)
sympy\sympy\integrals\meijerint.py:100:4: W0404: Reimport 'exp' (imported line 43) (reimported)
sympy\sympy\integrals\meijerint.py:100:4: W0404: Reimport 'sin' (imported line 47) (reimported)
sympy\sympy\integrals\meijerint.py:596:4: W0404: Reimport 'Heaviside' (imported line 42) (reimported)
sympy\sympy\integrals\meijerint.py:621:4: W0404: Reimport 'Wild' (imported line 38) (reimported)
sympy\sympy\integrals\meijerint.py:621:4: W0404: Reimport 'pi' (imported line 31) (reimported)
sympy\sympy\integrals\meijerint.py:621:4: W0404: Reimport 'oo' (imported line 31) (reimported)
sympy\sympy\integrals\meijerint.py:974:4: W0404: Reimport 'cos' (imported line 47) (reimported)
sympy\sympy\integrals\meijerint.py:974:4: W0404: Reimport 'exp' (imported line 43) (reimported)
sympy\sympy\integrals\meijerint.py:974:4: W0404: Reimport 'sin' (imported line 47) (reimported)
sympy\sympy\integrals\meijerint.py:1321:4: W0404: Reimport 'Or' (imported line 41) (reimported)
sympy\sympy\integrals\meijerint.py:1321:4: W0404: Reimport 'And' (imported line 41) (reimported)
sympy\sympy\integrals\meijerint.py:1321:4: W0404: Reimport 'exp' (imported line 43) (reimported)
sympy\sympy\integrals\meijerint.py:1321:4: W0404: Reimport 'Add' (imported line 34) (reimported)
sympy\sympy\integrals\meijerint.py:1467:4: W0404: Reimport 'oo' (imported line 31) (reimported)
sympy\sympy\integrals\meijerint.py:1467:4: W0404: Reimport 'Tuple' (imported line 29) (reimported)
sympy\sympy\integrals\meijerint.py:1535:4: W0404: Reimport 'oo' (imported line 1467) (reimported)
sympy\sympy\integrals\meijerint.py:1535:4: W0404: Reimport 'zoo' (imported line 1467) (reimported)
sympy\sympy\integrals\meijerint.py:1535:4: W0404: Reimport 'oo' (imported line 31) (reimported)
sympy\sympy\integrals\meijerint.py:1554:8: W0404: Reimport 'hyperexpand' (imported line 39) (reimported)
sympy\sympy\integrals\meijerint.py:1657:4: W0404: Reimport 'meijerg' (imported line 48) (reimported)
sympy\sympy\integrals\meijerint.py:1683:4: W0404: Reimport 'piecewise_fold' (imported line 44) (reimported)
sympy\sympy\integrals\meijerint.py:1802:4: W0404: Reimport 'exp' (imported line 43) (reimported)
sympy\sympy\integrals\meijerint.py:1802:4: W0404: Reimport 'And' (imported line 41) (reimported)
sympy\sympy\integrals\meijerint.py:1802:4: W0404: Reimport 'DiracDelta' (imported line 42) (reimported)
sympy\sympy\integrals\meijerint.py:2092:4: W0404: Reimport 'exp' (imported line 43) (reimported)
sympy\sympy\integrals\meijerint.py:2092:4: W0404: Reimport 'expand' (imported line 33) (reimported)
sympy\sympy\integrals\meijerint.py:2092:4: W0404: Reimport 'Add' (imported line 34) (reimported)
sympy\sympy\integrals\meijerint.py:2092:4: W0404: Reimport 'Mul' (imported line 35) (reimported)
sympy\sympy\integrals\meijerint.py:2092:4: W0404: Reimport 'Heaviside' (imported line 42) (reimported)
************* Module sympy.integrals.transforms
sympy\sympy\integrals\transforms.py:244:4: W0404: Reimport 'Max' (imported line 13) (reimported)
sympy\sympy\integrals\transforms.py:329:8: W0404: Reimport 'Max' (imported line 13) (reimported)
sympy\sympy\integrals\transforms.py:426:4: W0404: Reimport 'pi' (imported line 1655) (reimported)
sympy\sympy\integrals\transforms.py:426:4: W0404: Reimport 'gamma' (imported line 4) (reimported)
sympy\sympy\integrals\transforms.py:496:4: W0404: Reimport 'gamma' (imported line 4) (reimported)
sympy\sympy\integrals\transforms.py:496:4: W0404: Reimport 'exp' (imported line 4) (reimported)
sympy\sympy\integrals\transforms.py:496:4: W0404: Reimport 'pi' (imported line 1655) (reimported)
sympy\sympy\integrals\transforms.py:496:4: W0404: Reimport 'sin' (imported line 1655) (reimported)
sympy\sympy\integrals\transforms.py:496:4: W0404: Reimport 'cos' (imported line 1655) (reimported)
sympy\sympy\integrals\transforms.py:748:4: W0404: Reimport 'pi' (imported line 1655) (reimported)
sympy\sympy\integrals\transforms.py:748:4: W0404: Reimport 'gamma' (imported line 4) (reimported)
sympy\sympy\integrals\transforms.py:841:12: W0404: Reimport 'exp' (imported line 4) (reimported)
sympy\sympy\integrals\transforms.py:841:12: W0404: Reimport 'gamma' (imported line 4) (reimported)
sympy\sympy\integrals\transforms.py:841:12: W0404: Reimport 'sin' (imported line 1655) (reimported)
sympy\sympy\integrals\transforms.py:841:12: W0404: Reimport 'cos' (imported line 1655) (reimported)
sympy\sympy\integrals\transforms.py:1022:4: W0404: Reimport 'Max' (imported line 13) (reimported)
sympy\sympy\integrals\transforms.py:1022:4: W0404: Reimport 'exp' (imported line 4) (reimported)
sympy\sympy\integrals\transforms.py:1022:4: W0404: Reimport 'pi' (imported line 1655) (reimported)
sympy\sympy\integrals\transforms.py:1022:4: W0404: Reimport 'cos' (imported line 1655) (reimported)
sympy\sympy\integrals\transforms.py:1022:4: W0404: Reimport 'Wild' (imported line 4) (reimported)
sympy\sympy\integrals\transforms.py:1022:4: W0404: Reimport 'symbols' (imported line 4) (reimported)
sympy\sympy\integrals\transforms.py:1160:8: W0404: Reimport 'exp' (imported line 4) (reimported)
sympy\sympy\integrals\transforms.py:1164:8: W0404: Reimport 'Max' (imported line 13) (reimported)
sympy\sympy\integrals\transforms.py:1262:4: W0404: Reimport 'exp' (imported line 4) (reimported)
sympy\sympy\integrals\transforms.py:1262:4: W0404: Reimport 'Integral' (imported line 14) (reimported)
sympy\sympy\integrals\transforms.py:1371:8: W0404: Reimport 'exp' (imported line 4) (reimported)
sympy\sympy\integrals\transforms.py:1486:4: W0404: Reimport 'exp' (imported line 4) (reimported)
sympy\sympy\integrals\transforms.py:1523:8: W0404: Reimport 'exp' (imported line 4) (reimported)
************* Module sympy.integrals.rubi.constraints
sympy\sympy\integrals\rubi\constraints.py:11:4: W0404: Reimport 'SubstForAux' (imported line 11) (reimported)
sympy\sympy\integrals\rubi\constraints.py:11:4: W0404: Reimport 'PolyLog' (imported line 11) (reimported)
```

Reply:
> There are a few of these... (Partial results)

I think it would be worth fixing these and adding a check to CI somehow.